packages/core/src/components/dialog/multistepDialog.tsx

```diff
@@ -84,7 +91,13 @@ export class MultistepDialog extends AbstractPureComponent2<MultistepDialogProps
         isOpen: false,
     };
 
-    public state: IMultistepDialogState = INITIAL_STATE;
+    public state: IMultistepDialogState;
+
+    constructor(props: MultistepDialogProps) {
+        super(props);
+
+        this.state = this.getInitialIndexFromProps(props);
```

Comment:
I think this can actually stay in the class member initializer, no constructor necessary:

```ts
public state: IMultistepDialogState = this.getInitialIndexFromProps(this.props);
```